geopandas/plotting.py

```diff
@@ -8,8 +9,12 @@
 
 from distutils.version import LooseVersion
 
+import matplotlib
 from ._decorator import doc
 
+if TYPE_CHECKING:
+    import matplotlib.pyplot
+
```

Comment:
```suggestion
from ._decorator import doc

if TYPE_CHECKING:
    import matplotlib
    import matplotlib.pyplot

```

Any import of optional dependency needs to be within `TYPE_CHECKING` clause.

Reply:
I'm still getting that it's undefined on line 325 of `plotting.py` https://github.com/steveberdy/geopandas/runs/3067901925?check_suite_focus=true

Reply:
When I had it outside of the TYPE_CHECKING if statement, the error stated that matplotlib was not defined: https://github.com/steveberdy/geopandas/runs/3067879240?check_suite_focus=true. I'll check the setup and init files.

Reply:
I am not sure what is the issue here, I'd have to check. Moving import elsewhere unconditionally is a no-go as `matplotlib` is an optional dependency. GeoPandas must work in environments without it.

Reply:
I could remove type hints from the plotting, since the documentation already states what is needed.

Reply:
I removed the optional type hints. Now, it appears the only errors I'm getting in workflows don't involve any code that I changed directly.